drivers/display/display_renesas_ra.c

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_ra_display
+
+#include "display_renesas_ra.h"
+#include "r_glcdc.h"
+#include <zephyr/drivers/clock_control/renesas_ra_cgc.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(display_renesas_ra, CONFIG_DISPLAY_LOG_LEVEL);
+
+struct display_ra_config {
+	const struct pinctrl_dev_config *pincfg;
+	const struct gpio_dt_spec backlight_gpio;
+	const struct device *clock_dev;
+	struct clock_control_ra_subsys_cfg clock_glcdc_subsys;
+	uint16_t height;
+	uint16_t width;
+	uint32_t display_frame_size;
+	enum display_pixel_format pixel_format;
+};
+
+struct display_ra_data {
+	glcdc_instance_ctrl_t display_ctrl;
+	display_cfg_t display_fsp_cfg;
+};
+
+static int ra_display_write(const struct device *dev, const uint16_t x, const uint16_t y,
+			    const struct display_buffer_descriptor *desc, const void *buf)
+{
+	struct display_ra_data *data = dev->data;
+	const struct display_ra_config *config = dev->config;
+	pixel_t *buffer_write = NULL;
+	uint32_t pixel_index, buffer_index;
+	uint32_t display_size, l_hstride;
+	const pixel_t *buffer_data = (const pixel_t *)buf;
+
+	__ASSERT(desc->width <= desc->pitch, "Pitch is smaller than width");
+	__ASSERT((desc->pitch * BYTE_PER_PIXEL * desc->height) <= desc->buf_size,
+		 "Input buffer to small");
+	display_size = config->display_frame_size;
+
+	l_hstride = data->display_fsp_cfg.input[0].hstride;
+	buffer_write = (pixel_t *)data->display_fsp_cfg.input[0].p_base;
+
+	for (uint32_t row = 0; row < desc->height; row++) {
```

Comment:
Ok, a few things here. Please feel free to ignore this comment if I'm misunderstanding the hardware, these are just suggestions :) 

This looks like a display that uses a memory mapped buffer, and streams it to the display. If so, I'd suggest the following:
- Add code to detect if the x, y, and `desc` parameters describe a buffer the same size as the display buffer. If so, pass that buffer directly to hardware. The Kconfigs you have setup for LVGL mean that you will get buffers the size of the screen passed to this function, and using the buffer directly should improve performance
- Add a Kconfig of some kind to allow users to disable allocation of the `fb_background##id` buffer. If a user knows the hardware they are working with, they will probably want to just pass full sized framebuffers to the display, and never use the driver-allocated framebuffer, making this wasted memory.

Reply:
> Add code to detect if the x, y, and desc parameters describe a buffer the same size as the display buffer. If so, pass that buffer directly to hardware. The Kconfigs you have setup for LVGL mean that you will get buffers the size of the screen passed to this function, and using the buffer directly should improve performance

Done

> Add a Kconfig of some kind to allow users to disable allocation of the fb_background##id buffer. If a user knows the hardware they are working with, they will probably want to just pass full sized framebuffers to the display, and never use the driver-allocated framebuffer, making this wasted memory.

Done. I added RENESAS_RA_GLCDC_FB_NUM to handle driver buffer allocation